tests/unit/clients/test_http.py

```diff
@@ -119,7 +121,7 @@ async def mock_valid_token_set():
     assert resp == resp_text
 
     assert 1 == mock_refresh_token_called
-    request = mocked.requests[('get', conftest.API_URL)][0]
+    request = mocked.requests[('get', yarl.URL(conftest.API_URL))][0]
```

Comment:
This is the only bit left. If we have to keep something like it (since I'm guessing you don't want to refactor http client to make testing default header addition easier), can we do something like `request = mocked.requests.popitem()[1][0]` instead? I know it's not super pretty, but I think it should be the only item, and it seems weird to do requiring a whole library for one already not-so-pretty line.

Reply:
I'm not sure I get your suggestion.  requests is a dict.  so surely it's popitem()[0]?  but also that's very much making an assumption that there's exactly one item in the dict

Reply:
I guess I could test len first

Reply:
nm, I didn't know how popitem() works, apparently

Reply:
how's this?